### PR TITLE
Add incremental deployment.

### PR DESCRIPTION
**Pull request by LiquoriChris**

### CODE REVIEW

**Comment by NowinskiK** [comment]
Hi @LiquoriChris , thanks for doing this. Much appreciate.
Would you be able to add:
1) Documentation about this feature in README file
2) Unit test for this feature
I'm also happy if you want to contribute more to this project and make it more useful, having code & feature-parity with adftools (azure.datafactory.tools) wherever possible.

File: test/Set-StateFromService.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    Import-Module -Name $moduleManifestPath -Force -Verbose:$false
+}
+
+InModuleScope azure.synapse.tools {
+    $testHelperPath = $PSScriptRoot | Join-Path -ChildPath 'TestHelper'
+    Import-Module -Name $testHelperPath -Force
+    Describe 'Set-StateFromService' {
+        Context 'Run test when throws an error' {
+            BeforeAll {
+                BeforeAll {
```
Comment (NowinskiK) [inline test/Set-StateFromService.Tests.ps1:15]: Duplicated section

File: test/Get-StateFromService.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    Import-Module -Name $moduleManifestPath -Force -Verbose:$false
+}
+
+InModuleScope azure.synapse.tools {
+    $testHelperPath = $PSScriptRoot | Join-Path -ChildPath 'TestHelper'
+    Import-Module -Name $testHelperPath -Force
+    Describe 'Get-StateFromService' {
+        Context 'Run test when throws an error' {
+            BeforeAll {
+                BeforeAll {
```
Comment (NowinskiK) [inline test/Get-StateFromService.Tests.ps1:15]: Duplicated section

File: public/Publish-SynapseFromJson.ps1 [hunk 3]
```diff
@@ -113,12 +116,14 @@ function Publish-SynapseFromJson {
     Write-Host "Synapse Workspace:  $SynapseWorkspaceName";
     Write-Host "Location:           $Location";
     Write-Host "Stage:              $Stage";
+    Write-Host "Storage Account:    $StorageAccountName";
```
Comment (NowinskiK) [inline public/Publish-SynapseFromJson.ps1:119]: Let's move this param to `SynapsePublishOption`

File: private/DeploymentState.class.ps1 [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+class SynapseGlobalParam {
+    $type = "Object"
+    $value = $null
+
+    SynapseGlobalParam ($value) 
+    {
+        $this.value = $value
+    }
+
+}
```
Comment (NowinskiK) [inline private/DeploymentState.class.ps1:105]: Where is this needed? (SynapseGlobalParam class)

**Review by NowinskiK — CHANGES_REQUESTED** [review]
I added few comments

File: test/Set-StateFromService.Tests.ps1 (hunk 1 above)
Comment (LiquoriChris) [reply test/Set-StateFromService.Tests.ps1]: Remove duplicate.

File: test/Get-StateFromService.Tests.ps1 (hunk 2 above)
Comment (LiquoriChris) [reply test/Get-StateFromService.Tests.ps1]: Remove duplicate.

File: private/DeploymentState.class.ps1 (hunk 4 above)
Comment (LiquoriChris) [reply private/DeploymentState.class.ps1:92]: Removed unused class.

**Comment by LiquoriChris** [comment]
@NowinskiK I made some adjustment on how I get the storage account name from the deployment state class instead using $targetsynapse.

**Comment by LiquoriChris** [comment]
> @NowinskiK I made some adjustment on how I get the storage account name from the deployment state class instead using $targetsynapse.

@NowinskiK please kindly review.

**Comment by NowinskiK** [comment]
Yes, I will do. I remember about this PR, but I'm a bit busy right now.

**Comment by NowinskiK** [comment]
Thank you @LiquoriChris for your patience and contribution!

**Pull request merged**